### PR TITLE
remove --machines and related deprecated flags from code and docs

### LINKED ISSUE #10295: Fix kube-controller-manager usage example: remove --machines
(in usage & therefore man pages)

### PR DESCRIPTION
**Pull request by mikedanese**
Flag was dead. Kept around for compatibility. Nodes will register themselves by default now.

Fixes #10295

@lavalamp

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -134,7 +133,6 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.DeletingPodsBurst, "deleting-pods-burst", 10, "Number of nodes on which pods are bursty deleted in case of node failure. For more details look into RateLimiter.")
 	fs.IntVar(&s.RegisterRetryCount, "register-retry-count", s.RegisterRetryCount, ""+
 		"The number of retries for initial node registration.  Retry interval equals node-sync-period.")
-	fs.Var(&s.MachineList, "machines", "List of machines to schedule onto, comma separated.")
 	fs.MarkDeprecated("machines", "will be removed in a future version")
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/controllermanager.go]: Remove this line, too?

**Comment by lavalamp** [comment]
How long ago was this deprecated?

**Comment by mikedanese** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/commit/d0bcf953e9cc2bf541e3f4d7220f581ee23b76c1

May 22

**Comment by mikedanese** [comment]
cc @roberthbailey since you originally marked these flags deprecated a month ago. As per #10295, we would like to remove these now. Mind being a second reviewer?

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**